src/Templates/src/Microsoft.Maui.Templates.csproj

```diff
@@ -14,13 +14,41 @@
     <ContentTargetFolders>content</ContentTargetFolders>
     <!-- This project has no .NET assemblies, so disable the warning for that -->
     <NoWarn>$(NoWarn);NU5128</NoWarn>
+    <NoBuild>true</NoBuild>
   </PropertyGroup>
 
   <ItemGroup>
     <Content Include="templates\**\*" Exclude="templates\**\bin\**;templates\**\obj\**" />
     <Compile Remove="**\*" />
   </ItemGroup>
 
+  <PropertyGroup>
+    <BeforePack>_UpdateTemplateVersions</BeforePack>
+  </PropertyGroup>
+
+  <!-- Eventually replaced by eng/Version.targets -->
+  <Target Name="SetVersions" />
+
+  <Target Name="_UpdateTemplateVersions" DependsOnTargets="SetVersions">
+    <ItemGroup>
+      <_TemplateJsonFile Include="templates\*\.template.config\template.json" />
+      <_TemplateJsonFileWithContent
+          Include="@(_TemplateJsonFile)"
+          Contents="$([System.IO.File]::ReadAllText('%(FullPath)')
+              .Replace('MICROSOFT_EXTENSIONS_VERSION_VALUE', '$(MicrosoftExtensionsFileProvidersEmbeddedPackageVersion)')
+              .Replace('PROJECT_REUNION_VERSION_VALUE', '$(_MicrosoftProjectReunionVersion)'))" />
```

Comment:
The hackiest way, but basically replace the values in the json files with the props values.